src/web/mod.rs

```diff
@@ -141,18 +141,35 @@ impl CratesfyiHandler {
 
 impl Handler for CratesfyiHandler {
     fn handle(&self, req: &mut Request) -> IronResult<Response> {
+        let mut response_time = metrics::ResponseTimer::new();
+        let mut response_route = metrics::ResponseRouter::new();
+
         // try serving shared rustdoc resources first, then router, then db/static file handler
         // return 404 if none of them return Ok
         self.shared_resource_handler
             .handle(req)
-            .or_else(|e| self.router_handler.handle(req).or(Err(e)))
+            .or_else(|e| {
+                let router = self.router_handler.handle(req).or(Err(e));
+
+                response_time.route("router");
+                response_route.route("router");
```

Comment:
This looks a little odd to me ... could you instead make `response_time` a field of `response_route` and have `response_route` call `response_time.route()` internally? That would prevent people from calling one but not the other by accident.

Reply:
I was considering making one struct that contained both sub-structs for this very reason, but I don't know if I should do that or just combine them outright because I can see situations where only one may apply, thoughts?

Reply:
I don't know when we'd want to know the route but not the response time. Having the response time separate sounds useful, but if we don't have a use case yet then I'd rather combine them for now.